madoop/mapreduce.py

```diff
@@ -173,7 +175,13 @@ def is_executable(exe):
             stderr=subprocess.PIPE,
             check=True,
         )
-    except (subprocess.CalledProcessError, OSError) as err:
+    except subprocess.CalledProcessError as err:
+        raise MadoopError(
+            f"Failed executable test: {err}\n"
+            f"{err.stdout.decode()}" if err.stdout else ""
```

Comment:
For each of these, there will be a new line even if there's no error output. Should these newlines be moved inside the f-strings to prevent that?

Reply:
Good suggestion.  Does this look like what you were thinking? 30214c9511b8c4185f251bee0dbd66f556c69a96

Reply:
Yep, looks good. Should line 182 also have it's own new line to separate the stderr output?

Reply:
stdout and stderr tend to include newlines already, so maybe it's not needed in this case.